src/librustc_trait_selection/traits/error_reporting/mod.rs

```diff
@@ -653,7 +653,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             }
 
             // Already reported in the query.
-            ConstEvalFailure(ErrorHandled::Reported) => {
+            ConstEvalFailure(ErrorHandled::Reported(ErrorReported) | ErrorHandled::Linted) => {
```

Comment:
On `::Linted`, we might still do `delay_span_bug`... so an ICE is possible?

Reply:
Maybe, but I don't think this behavior is changed in this PR.

Reply:
Sure; could you file an issue or leave a comment though?

Reply:
Oh wait this is the trait system not codegen. I'll add a `delay_span_bug`.

Reply:
Wait, no, `delay_span_bug` is there, that was your point. Argh, GitHub!

Reply:
AFAIK linting doesn't happen on any of the constants that could show up here. `delay_span_bug` here will catch if that happens. I will  split it thought, so it's clearer what's going on.